pilot/pkg/networking/v1alpha3/gateway.go

```diff
@@ -16,161 +16,168 @@ package v1alpha3
 
 import (
 	"fmt"
-	"path"
-	"strconv"
-	"strings"
 
 	xdsapi "github.com/envoyproxy/go-control-plane/envoy/api/v2"
-	"github.com/envoyproxy/go-control-plane/envoy/api/v2/auth"
-	"github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
 	http_conn "github.com/envoyproxy/go-control-plane/envoy/config/filter/network/http_connection_manager/v2"
 	"go.uber.org/zap"
 
+	"github.com/envoyproxy/go-control-plane/envoy/api/v2/route"
+
+	"github.com/envoyproxy/go-control-plane/envoy/api/v2/auth"
+	"github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
+	"github.com/gogo/protobuf/types"
+
 	networking "istio.io/api/networking/v1alpha3"
 	"istio.io/istio/pilot/pkg/model"
-	deprecated "istio.io/istio/pilot/pkg/proxy/envoy/v1"
 	"istio.io/istio/pkg/log"
 )
 
-func buildGatewayHTTPListeners(env model.Environment, node model.Proxy) ([]*xdsapi.Listener, error) {
+func buildGatewayListeners(env model.Environment, node model.Proxy) ([]*xdsapi.Listener, error) {
 
 	config := env.IstioConfigStore
 
-	gateways, err := config.List(model.Gateway.Type, model.NamespaceAll)
+	// collect workload labels
+	workloadInstances, err := env.GetProxyServiceInstances(node)
 	if err != nil {
-		return nil, fmt.Errorf("listing gateways: %s", err)
+		log.Error("Failed to get gateway instances for router ", zap.String("node", node.ID), zap.Error(err))
```

Comment:
Can you use Errora - and not depend directly on zap ? 

(can fix it in separate PR )

Reply:
I just copied the pattern from other files.. will fix